From ec5c9f0548e9a2804040a6ce81490667842c564e Mon Sep 17 00:00:00 2001
From: Laura Abbott <labbott@redhat.com>
Date: Wed, 19 Jun 2019 16:28:21 -0400
Subject: [PATCH 32/35] Make get_cert_list() use efi_status_to_str() to print
 error messages.

Message-id: <20190619162824.427-17-labbott@redhat.com>
Patchwork-id: 265694
O-Subject: [ARK PATCH 32/35] Make get_cert_list() use efi_status_to_str() to print error messages.
Bugzilla:
RH-Acked-by: Jarod Wilson <jarod@redhat.com>

From: Peter Jones <pjones@redhat.com>

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 security/integrity/platform_certs/load_uefi.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)
Signed-off-by: Laura Abbott <labbott@redhat.com>
---
 security/integrity/platform_certs/load_uefi.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/security/integrity/platform_certs/load_uefi.c b/security/integrity/platform_certs/load_uefi.c
index e188f3ecbce3..4241accc740e 100644
--- a/security/integrity/platform_certs/load_uefi.c
+++ b/security/integrity/platform_certs/load_uefi.c
@@ -54,7 +54,8 @@ static __init int get_cert_list(efi_char16_t *name, efi_guid_t *guid,
 	}
 
 	if (status != EFI_BUFFER_TOO_SMALL) {
-		pr_err("Couldn't get size: 0x%lx\n", status);
+		pr_err("Couldn't get size: %s (0x%lx)\n",
+		       efi_status_to_str(status), status);
 		return efi_status_to_err(status);
 	}
 
@@ -65,7 +66,8 @@ static __init int get_cert_list(efi_char16_t *name, efi_guid_t *guid,
 	status = efi.get_variable(name, guid, NULL, &lsize, db);
 	if (status != EFI_SUCCESS) {
 		kfree(db);
-		pr_err("Error reading db var: 0x%lx\n", status);
+		pr_err("Error reading db var: %s (0x%lx)\n",
+		       efi_status_to_str(status), status);
 		return efi_status_to_err(status);
 	}
 
-- 
2.21.0

